hoot-core/src/main/cpp/hoot/core/io/OsmApiChangesetElement.h

```diff
@@ -394,54 +395,48 @@ class ElementIdToIdMap
    */
   void updateId(ElementType::Type type, long old_id, long new_id)
   {
-    _oldIdToNew[type][old_id] = new_id;
-    _newIdToOld[type][new_id] = _newIdToOld[type][old_id];
-    _newIdToOld[type].erase(old_id);
+    _idToId[type][old_id] = new_id;
+    _idToId[type][new_id] = old_id;
   }
   /**
-   * @brief getOldId Get the old ID from the new ID
+   * @brief getId Get the new or old ID from the ID passed in
    * @param type Element type (node/way/relation)
-   * @param new_id Updated (new) ID
-   * @return old ID
+   * @param id ID
+   * @return new or old ID
    */
-  long getOldId(ElementType::Type type, long new_id)
+  long getId(ElementType::Type type, long id)
   {
-    if (_newIdToOld[type].find(new_id) == _newIdToOld[type].end())
-      return new_id;
+    if (_idToId[type].find(id) == _idToId[type].end())
+      return id;
     else
-      return _newIdToOld[type][new_id];
+      return _idToId[type][id];
   }
   /**
-   * @brief getNewId Get the new ID from the old ID
-   * @param type Element type (node/way/relation)
-   * @param old_id Original (old) ID
-   * @return new ID
+   * @brief containsId
+   * @param type
+   * @param id
+   * @return
    */
-  long getNewId(ElementType::Type type, long old_id)
+  bool containsId(ElementType::Type type, long id)
   {
-    if (_oldIdToNew[type].find(old_id) == _oldIdToNew[type].end())
-      return old_id;
-    else
-      return _oldIdToNew[type][old_id];
+    return _idToId[type].find(id) != _idToId[type].end();
   }
   /**
    * @brief begin Get beginning iterator
    * @param type Element type (node/way/relation)
    * @return begin iterator
    */
-  iterator begin(ElementType::Type type)  { return _newIdToOld[type].begin(); }
+  iterator begin(ElementType::Type type)  { return _idToId[type].begin(); }
   /**
    * @brief end Get ending iterator
    * @param type Element type (node/way/relation)
    * @return end iterator
    */
-  iterator end(ElementType::Type type)    { return _newIdToOld[type].end(); }
+  iterator end(ElementType::Type type)    { return _idToId[type].end(); }
 
 private:
   /** Old ID to new ID mapping */
-  XmlElementIdMap _oldIdToNew;
-  /** New ID to old ID mapping */
-  XmlElementIdMap _newIdToOld;
+  ChangesetElementIdMap _idToId;
```

Comment:
Another large memory overhead was the two element ID maps.  One for `old -> new` mappings and the other `new -> old` mappings.  Now they are combined in the same map.